src/components/FileUploaderV2/FileUploaderV2.js

```diff
@@ -0,0 +1,285 @@
+    action: PropTypes.string,
+    upload: PropTypes.func,
+  };
+
+  static defaultProps = {
+    iconDescription: 'Provide icon description',
+    buttonLabel: 'Upload',
+    buttonKind: 'primary',
+    multiple: false,
+    onClick: () => {},
+    onChange: () => {},
+    accept: [],
+    upload: () => {},
+  };
+
+  state = {
+    files: [],
+  };
+
+  nodes = [];
+
+  handleChange = evt => {
+    evt.stopPropagation();
+    const files = [...this.state.files];
+    [...evt.target.files].forEach(file => {
+      const uuid = uid();
+      files.push({
+        uuid,
+        name: file.name,
+        size: file.size,
+        status: 'uploading',
+      });
+      const index = files.findIndex(file => file.uuid === uuid);
+      this.props
+        .upload({ file })
+        .then(() => {
+          files[index].status = 'complete';
+          this.setState({ files });
+        })
+        .catch(error => {
+          files[index].status = 'edit';
+          this.setState({ files });
+          return new Error(error);
+        });
+    });
+    this.setState({ files });
+    this.props.onChange(evt);
+  };
+
+  handleClick = ({ evt, index }) => {
+    const filteredArray = this.state.files.filter(
+      file => file.name !== this.nodes[index].innerText.trim()
+    );
+    this.setState({ files: filteredArray });
+    this.props.onClick(evt);
+  };
+
+  clearFiles = () => {
+    // A clearFiles function that resets filenames and can be referenced using a ref by the parent.
+    this.setState({ files: [] });
+  };
+
+  render() {
+    const {
+      iconDescription,
+      buttonLabel,
+      buttonKind,
+      labelDescription,
+      labelTitle,
+      className,
+      multiple,
+      accept,
+      name,
+      upload, // eslint-disable-line
+      ...other
+    } = this.props;
+
+    const classes = classNames({
+      'bx--form-item': true,
+      [className]: className,
+    });
+
+    return (
+      <div className={classes} {...other}>
+        <strong className="bx--label">{labelTitle}</strong>
+        <p className="bx--label-description">{labelDescription}</p>
+        <FileUploaderButtonV2
+          labeltext={buttonLabel}
+          multiple={multiple || undefined}
+          buttonKind={buttonKind}
+          onChange={this.handleChange}
+          accept={accept}
+          name={name}
+        />
+        <div className="bx--file-container">
+          {this.state.files.length
+            ? this.state.files.map((file, index) => (
+                <span
+                  key={uid()}
+                  className="bx--file__selected-file"
+                  ref={node => (this.nodes[index] = node)}
+                  {...other}>
+                  <p className="bx--file-filename">{file.name}</p>
+                  <span className="bx--file__state-container">
+                    <FilenameV2
+                      iconDescription={iconDescription}
+                      status={file.status}
```

Comment:
Seems that `filenameStatus` is not used in `<FileUploaderV2>` but still I see `filenameStatus` in https://github.com/emyarod/carbon-components-react/blob/9576d6b/src/components/FileUploaderV2/FileUploaderV2-story.js#L68. This seems to be causing React’s unknown prop warning.

Reply:
I've changed the `filenameStatus` and `labelText` props to all lowercase which should remove the console warnings

Reply:
Thanks @emyarod for addressing this! `filenameStatus` to `status` looks a good change. Could you please explain why the change from `labelText` to `labeltext` addresses the console warnings?

Reply:
React wants to differentiate DOM-related props from [custom props](https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html), so they try to [keep DOM-related props in camelCase and custom props in lowercase](https://reactjs.org/docs/dom-elements.html#all-supported-html-attributes)

This is the warning that is given if you pass a custom camelCase attribute

<img width="502" alt="screen shot 2018-07-26 at 8 09 00 am" src="https://user-images.githubusercontent.com/8265238/43264279-3b9db424-90ab-11e8-8b74-9aa73ff2dec9.png">

Reply:
Thank you for explaining @emyarod! This is caused by `labelText` going to `<label>`’s attribute in addition to `<label>`’s children, caused by `{…other}` thing. Basically, what you need to do is listing everything that goes other than `<label>`’s attribute in [here](https://github.com/emyarod/carbon-components-react/blob/67f26d3/src/components/FileUploaderV2/FileUploaderV2.js#L35-L42).

Reply:
got it @asudoh, I've reverted the prop name change and am now explicitly declaring the prop instead of spreading it like before